### PR TITLE
Bees and other venomous mobs properly inject their venom.

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About The Pull Request
Adds exposure handling to bee stings and the venom element. 

## Why It's Good For The Game
This mostly just makes sense as a consistency thing. You have a mob, that mob is biting/stinging/whatevering you, and as a result toxin is being transferred from the mob to your bloodstream. It has to get in somehow and in most other cases of reagent-transfer-to-mob the mob is exposed to the reagent on transfer. There's no reason for venom to not.

Might incentivize some creativity with medical bee setups or something.

## Changelog

:cl:
add: Bees and other venomous mobs properly inject their venom.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/living_defense.dm [hunk 1]
```diff
@@ -501,7 +501,8 @@
 		taste(source)
 
 	var/touch_protection = (methods & VAPOR) ? get_permeability_protection() : 0
-	SEND_SIGNAL(source, COMSIG_REAGENTS_EXPOSE_MOB, src, reagents, methods, volume_modifier, show_message, touch_protection)
+	if (source)
```
Comment (Fikou) [inline code/modules/mob/living/living_defense.dm:504]: why?

Comment (TemporalOroboros) [reply code/modules/mob/living/living_defense.dm:504]: Bees and venom don't have a source holder. Trying to send a signal to null runtimes.

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 2]
```diff
@@ -179,10 +179,22 @@
...
+ * Arguments:
+ * - [target][/mob/living]: The mob that has been stung.
+ */
+/mob/living/simple_animal/hostile/bee/proc/on_sting(mob/living/target)
+	var/sting_amt = rand(1, 5)
+
+	target.reagents?.add_reagent(beegent.type, sting_amt)
+	target.expose_reagents(list((beegent) = sting_amt), null, INJECT, volume_modifier = 1, show_message = TRUE)
+	if (target.reagents)
+		beegent.on_transfer(target, INJECT, sting_amt)
```
Comment (ATH1909) [inline code/modules/mob/living/simple_animal/hostile/bees.dm:196]: why are you adding the reagent, then transferring it?

Comment (TemporalOroboros) [reply code/modules/mob/living/simple_animal/hostile/bees.dm:196]: That's on_transfer, not trans_to. It handles whatever effects the reagent has upon being transferred to something else.

**Comment by tralezab** [comment]
conflict

File: code/datums/elements/venomous.dm [hunk 3]
```diff
@@ -54,9 +58,15 @@
...
+
 	var/final_amount_added
 	if(islist(amount_added))
 		final_amount_added = rand(amount_added[1], amount_added[2])
 	else
 		final_amount_added = amount_added
-	target.reagents?.add_reagent(poison_type, final_amount_added)
+
+	target.reagents?.add_reagent(venom.type, final_amount_added)
+	if (methods)
```
Comment (SuperNovaa41) [inline code/datums/elements/venomous.dm:69]: do you need to check this? methods is set to `TOUCH` if no value is provided

File: code/datums/components/fantasy/prefixes.dm [hunk 4]
```diff
@@ -137,7 +137,7 @@
 		/datum/reagent/toxin/histamine = "creeping malaise"
 	)
 	var/poisonname = names[picked_poison]
-	master.AddElement(/datum/element/venomous, picked_poison, comp.quality+1)
+	master.AddElement(/datum/element/venomous, picked_poison, comp.quality+1, INJECT)
```
Comment (SuperNovaa41) [inline code/datums/components/fantasy/prefixes.dm:140]: You should be specifying the argument for this, see: https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#prefer-named-arguments-when-the-meaning-is-not-obvious

File: code/datums/elements/venomous.dm (hunk 3 above)
Comment (TemporalOroboros) [reply code/datums/elements/venomous.dm:69]: Someone might pass NONE so it kind of needs to.

File: code/modules/mob/living/simple_animal/hostile/bees.dm (hunk 2 above)
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/hostile/bees.dm:196]: Calling this proc is managed by the `trans_to` proc normally and using it externally for a particular feature is a bad sign. If you need this to be called you should be moving the reagents using the usual procs.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]
I may be misreading this pr a bit but it looks to me like you're trying to work around having a source when it would be easier and less hacky to just manage a reagents source from inside the component in the first place.

**Pull request closed without merging**